### PR TITLE
Igor/fix ci

### PR DESCRIPTION
**Pull request by ig-ra**
fix: CI (go cross-platform releaser die to CGO=1)

Fixes ENG-356 (https://linear.app/autokitteh/issue/ENG-356)

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -1,31 +1,103 @@
...
+    flags:
+      - -mod=readonly
+    ldflags:
+      - -s -w -X main.version={{.Version}}
 
 source:
   enabled: false
 
 release:
+#  github:
```
Comment (itayd) [inline .goreleaser.yaml]: kill?

File: .goreleaser.yaml [hunk 2]
```diff
@@ -1,31 +1,103 @@
...
-      {{ .ProjectName }}_
+      {{ .Binary }}_
       {{- title .Os }}_
       {{- if eq .Arch "amd64" }}x86_64
       {{- else if eq .Arch "386" }}i386
       {{- else }}{{ .Arch }}{{ end }}
       {{- if .Arm }}v{{ .Arm }}{{ end }}
+
+    format_overrides:
+      - goos: windows
```
Comment (itayd) [inline .goreleaser.yaml]: i thought we're not supporting windows?

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/autokitteh/starlark-lsp
 
-go 1.19
+go 1.21
```
Comment (itayd) [inline go.mod:3]: why is this needed?

**Comment by itayd** [comment]
do you have an example output to show?

**Comment by ig-ra** [comment]
@itayd example outout? you mean release?

File: go.mod (hunk 3 above)
Comment (ig-ra) [reply go.mod:3]: aligned with autokitteh. Do you want me to leave this 1.19?

File: .goreleaser.yaml (hunk 2 above)
Comment (ig-ra) [reply .goreleaser.yaml]: added windows support

File: go.mod (hunk 3 above)
Comment (ig-ra) [reply go.mod:3]: I prefer to state it explicitly. We need go version for goreleaser as well (see Makefile)

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -10,26 +10,21 @@ on:
...
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Set up Go
+      - name: setup dependencies
         uses: actions/setup-go@v4
         with:
           cache-dependency-path: ${{env.WORKING_DIRECTORY}}/go.sum
           go-version-file: ${{env.WORKING_DIRECTORY}}/go.mod
```
Comment (efiShtain) [inline .github/workflows/release.yml]:
```suggestion
          cache-dependency-path: go.sum
          go-version-file: go.mod
```

and remove the working_directory env, you don't need it
we use it an ak since we have multiple libraries in one repo

File: Makefile [hunk 5]
```diff
@@ -23,7 +23,22 @@ fmt:
...
 	go run ./hack/starlark-builtins.go > pkg/analysis/builtins.py
+
+
+.PHONY: sysroot-pack sysroot-unpack release-dry-run release
+
+PACKAGE_NAME          := github.com/autokitteh/autokitteh-starlark-lsp
+GOLANG_CROSS_VERSION  ?= v1.21  # could use latest, but it's better to specify exact Go version
+
+DOCKER_RUN = docker run --rm -e CGO_ENABLED=1 -v /var/run/docker.sock:/var/run/docker.sock -v `pwd`:/go/src/$(PACKAGE_NAME) -v `pwd`/sysroot:/sysroot -w /go/src/$(PACKAGE_NAME)
+GORELEASER_IMAGE = ghcr.io/goreleaser/goreleaser-cross:${GOLANG_CROSS_VERSION}
```
Comment (efiShtain) [inline Makefile:39]: add a space above please 
seems like a one liner but it's two different variables

File: .goreleaser.yaml [hunk 6]
```diff
@@ -1,16 +1,98 @@
...
+      - arm
+    goarm:
+      - 7
+    env:
+      - CC=arm-linux-gnueabihf-gcc
+      - CXX=arm-linux-gnueabihf-g++
+      - CGO_CFLAGS=--sysroot=/sysroot/linux/armhf
+      - CGO_LDFLAGS=--sysroot=/sysroot/linux/armhf
+      - PKG_CONFIG_SYSROOT_DIR=/sysroot/linux/armhf
+      - PKG_CONFIG_PATH=/sysroot/linux/armhf/opt/vc/lib/pkgconfig:/sysroot/linux/armhf/usr/lib/arm-linux-gnueabihf/pkgconfig:/sysroot/linux/armhf/usr/lib/pkgconfig:/sysroot/linux/armhf/usr/local/lib/pkgconfig
```
Comment (efiShtain) [inline .goreleaser.yaml]: this assumes you run this on a specific machine with specific arch ?
maybe a comment will be helpful here

File: Makefile [hunk 7]
```diff
@@ -23,7 +23,22 @@ fmt:
...
+
+release-dry-run:
+	@$(DOCKER_RUN) $(GORELEASER_IMAGE) --clean --skip=validate --skip=publish --debug --verbose
+
+release:
```
Comment (efiShtain) [inline Makefile:44]: don't you need a release-mac feature for testing instead of releasing all platforms every time ?

File: Makefile [hunk 8]
```diff
@@ -23,7 +23,22 @@ fmt:
...
 builtins:
 	go run ./hack/starlark-builtins.go > pkg/analysis/builtins.py
+
+
+.PHONY: sysroot-pack sysroot-unpack release-dry-run release
+
+PACKAGE_NAME          := github.com/autokitteh/autokitteh-starlark-lsp
+GOLANG_CROSS_VERSION  ?= v1.21  # could use latest, but it's better to specify exact Go version
+
+DOCKER_RUN = docker run --rm -e CGO_ENABLED=1 -v /var/run/docker.sock:/var/run/docker.sock -v `pwd`:/go/src/$(PACKAGE_NAME) -v `pwd`/sysroot:/sysroot -w /go/src/$(PACKAGE_NAME)
```
Comment (efiShtain) [inline Makefile:37]: This won't work (mounting the docker socket) if we move to our hosted gitlab runners (which we already considered before) unless we will allow privileged access on the runners (which we won't)

so fine for now as long as GitHub allows it, but keep a mental note we might need to change in the future

File: .goreleaser.yaml (hunk 6 above)
Comment (ig-ra) [reply .goreleaser.yaml]: no, it's a cross compiling. docker is the same docker. But there are several compilers and related things installed. 
The underlying arch of the docker is linux amd64, unrelated to target arch

File: Makefile (hunk 7 above)
Comment (ig-ra) [reply Makefile:44]: please elaborate?

Comment (efiShtain) [reply Makefile:44]: not a must
but assume you want to create versions on a dev branch and let people test it
so release only a specific arch might be faster and waste less action minuets

File: Makefile (hunk 8 above)
Comment (ig-ra) [reply Makefile:37]: I think there will be no problems once we decide to change it.
docker should be able to run and publish to github.

Comment (efiShtain) [reply Makefile:37]: the problem is with mounting the docker socket and talking to the host
it requires root privileges on the host 

but for now, if it works, fine by me

File: Makefile (hunk 7 above)
Comment (ig-ra) [reply Makefile:44]: - there is nothing platform-specific in LSP code, so I see no need to release for specific platform.
- moreover, there won't be frequent releases, therefore I don't really see a reason why to make releases faster and what will be the use case for this.

File: Makefile (hunk 8 above)
Comment (ig-ra) [reply Makefile:37]: it works for now, yes. But if you have a better solution, please suggest. The only assumption now that we need a docker with cross-compilers installed.

File: Makefile (hunk 5 above)
Comment (ig-ra) [reply Makefile:39]: added

File: .github/workflows/release.yml (hunk 4 above)
Comment (ig-ra) [reply .github/workflows/release.yml]: removed

**Review by itayd — APPROVED** [review]
